src/feature/control/control_events.c

```diff
@@ -1318,6 +1318,16 @@ control_event_logmsg(int severity, log_domain_mask_t domain, const char *msg)
       for (cp = b; *cp; ++cp)
         if (*cp == '\r' || *cp == '\n')
           *cp = ' ';
+
+      /* Remove trailing spaces */
+      for (--cp; *cp == ' ' && cp >= b; --cp)
+        *cp = '\0';
+
+      if ( cp == b ){
```

Comment:
The `cp == b` check isn't necessary, `tor_assert_nonfatal(*b);` makes sure that b is not an empty string.